contracts/rewards/src/state.rs

```diff
@@ -108,17 +124,12 @@ impl RewardsPool {
         }
     }
 
-    pub fn distribute_rewards(
-        &mut self,
-        worker_count: u64,
-        rewards_per_worker: Uint128,
-    ) -> Result<(), ContractError> {
-        let rewards = rewards_per_worker * Uint128::from(worker_count as u128);
-        if self.balance < rewards {
+    pub fn sub_reward(mut self, reward: Uint128) -> Result<Self, ContractError> {
+        if self.balance < reward {
             return Err(ContractError::PoolBalanceInsufficient.into());
         }
-        self.balance -= rewards;
-        Ok(())
+        self.balance -= reward;
+        Ok(self)
```

Comment:
```suggestion
        
        Ok(self)
```
Prefer always having an empty line before return.